pybossa/view/projects.py

```diff
@@ -3215,19 +3228,35 @@ def coowners(short_name):
             # delete ids that don't exist anymore
             delete = [value for value in old_list if value not in overlap_list]
             for _id in delete:
+                # Remove the coowner.
                 project.owners_ids.remove(_id)
             # add ids that weren't there
             add = [value for value in new_list if value not in overlap_list]
             for _id in add:
                 project.owners_ids.append(_id)
 
+            # Update coowners_dict in response.
+            coowners_dict = [{"id": user.id, "fullname": user.fullname} for user in user_repo.get_users(project.owners_ids)]
+            response['coowners_dict'] = coowners_dict
+
             # save contacts
             new_list = [int(x) for x in json.loads(request.data).get('contacts', [])]
+
+            # remove any contacts that were just removed from coowners.
+            for _id in delete:
+                if _id in new_list:
+                    new_list.remove(_id)
+
             auditlogger.log_event(project, current_user, 'update', 'project.contacts',
                 project.info.get('contacts'), new_list)
             project.info['contacts'] = new_list
 
             project_repo.save(project)
+
+            # Update contacts_dict in response.
```

Comment:
line 3228 - line 3240 also update coowners_dict. Are they duplicate?

Reply:
The two are different, as one updates `contacts_dict` in the `response` object, while the other `coowners_dict`. This was previously not updated, but the unit tests identified that the fields should be changed to be in-sync.